### PR TITLE
Add section about commit messages

### PR DESCRIPTION
**Pull request by MarkoSagadin**
Consider this PR as work in progress, The sections `Documentation in source code` and `Coding Standards` are just placeholders for now.

Please read the linked resources about commit guidelines.
I would like to only highlight critical things in this section, as I think that the first resource is well written and there is no need to reinvent the wheel.

Write if you would add anything (section about `Message content` needs to be written), I will also mention `gitlint` tool.

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
Zephyr's commit guidelines have few interesting guidelines (not talking about style here, but other stuf): https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines

File: docs/developer_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    * [Message style](#message-style)
+    * [Message content](#message-content)
+* [Documentation in source code 📑](#documentation-in-source-code-)
+* [Coding standards 📚](#coding-standards-)
+    * [C/C++ 🇨](#cc-)
+    * [Python 🐍](#python-)
+    * [Rust 🦀](#rust-)
+    * [JavaScript/TypeScript 🇯🇸 / 🇹🇸](#javascripttypescript---)
+
+<!-- vim-markdown-toc -->
```
Comment (TjazVracko) [inline docs/developer_guidelines.md:17]: image

The `C` emoji is barely visible, 
Google chrome v100.0.4896.127 doesnt display the `JS/TS` emojis

**Comment by MarkoSagadin** [comment]
Tjaz comments: we should write full messages instead of short ones even in the cases where there is a github issue that describes the problem. We still refernce the issue and provide context.

Above comment refers to below section:
> Short commit message should be around 50 characters long, maximum is at 72.
>
> Use this when the change that you made is either:
> * self-explanatory - For example: `Fix typo in README.md` or
> * there is a GitHub issue that provides full context for the change made, for example: `Fix out of bounds bug, closes #123`

File: docs/developer_guidelines.md (hunk 1 above)
Comment (vid553) [reply docs/developer_guidelines.md:17]: Firefox 100.0 does not have any problems displaying it.

File: docs/developer_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+You can also include bullet points:
+* A
+* B
+* C
+
+Reference GitHub issues at the bottom, like this:
+
+Resolves: #123
+See also: #456, #789
```
[the remark this answers is not in the document]
Comment (MarkoSagadin) [reply docs/developer_guidelines.md:72]: Agreed, refer to the "Short" subsection.

**Review by NejcKle — APPROVED** [review]

File: docs/developer_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+#### Short
+
+Short commit messages should be around 50 characters long, the maximum is 72.
+
+Use this when the change that you made is either:
+* self-explanatory, for example: `Fix typo in README.md` or
+* there is a GitHub issue that provides sufficient context for the change made.
+
+Mention relevant GitHub issues at the end of the message, for example:
+* `Fix out of bounds bug, closes #123`
```
Comment (phillipjohnston) [inline docs/developer_guidelines.md:52]: Does this actually work to close the issue? I thought GitHub required it to be on its own line, but maybe that is not true anymore!

Comment (MarkoSagadin) [reply docs/developer_guidelines.md:52]: Yep, it does! It does not matter where in the commit message the issue is mentioned. But to close the issue with the `Closes` keyword the commit should be pushed to the branch set as a default.

If master is set as default and you are regularly committing to the dev, the issues will only be closed after the merge from dev into master is done.

**Pull request merged**